src/types/window.ts

```diff
@@ -0,0 +1,10 @@
+import { MetaMaskInpageProvider } from '@metamask/providers'
+
+declare global {
+  interface Window {
+    ethereum?: MetaMaskInpageProvider
```

Comment:
Is it possible that in a client's website `ethereum` not be the `MetaMaskInpageProvider` but some other provider?

Reply:
As I know- no. I didn't meet any other provider under `ethereum` object in my experience